pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+	namespace     string
+	secretName    string
+	canWatchNodes bool
+}
+
+type nodeInfo struct {
+	imNodes map[string]string
+	nodes   []corev1.Node
+}
+
+func NewBootstrapperInitGenerator(client client.Client, apiReader client.Reader, dtClient dtclient.Client, namespace string) *BootstrapperInitGenerator {
+	return &BootstrapperInitGenerator{
+		client:     client,
+		dtClient:   dtClient,
+		apiReader:  apiReader,
+		namespace:  namespace,
+		secretName: consts.BootsTrapperInitSecretName,
+	}
+}
+
+// GenerateForDynakube creates/updates the init secret for EVERY namespace for the given dynakube.
+// Used by the dynakube controller during reconcile.
+func (g *BootstrapperInitGenerator) GenerateForDynakube(ctx context.Context, dk *dynakube.DynaKube) error {
+	log.Info("reconciling namespace bootstrapper init secret for", "dynakube", dk.Name)
+
+	g.canWatchNodes = true
+
+	data, err := g.generate(ctx, dk)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	nsList, err := mapper.GetNamespacesForDynakube(ctx, g.apiReader, dk.Name)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	coreLabels := k8slabels.NewCoreLabels(dk.Name, k8slabels.WebhookComponentLabel)
+
+	secret, err := k8ssecret.BuildForNamespace(consts.BootsTrapperInitSecretName, "", data, k8ssecret.SetLabels(coreLabels.BuildLabels()))
+	if err != nil {
+		return err
+	}
+
+	err = k8ssecret.Query(g.client, g.apiReader, log).CreateOrUpdateForNamespaces(ctx, secret, nsList)
+	if err != nil {
+		return err
+	}
+
+	log.Info("done updating init secrets")
+
+	return nil
+}
+
+func (g *BootstrapperInitGenerator) Cleanup(ctx context.Context, namespaces []corev1.Namespace) error {
+	nsList := make([]string, 0, len(namespaces))
+	for _, ns := range namespaces {
+		nsList = append(nsList, ns.Name)
+	}
+
+	return k8ssecret.Query(g.client, g.apiReader, log).DeleteForNamespaces(ctx, consts.BootsTrapperInitSecretName, nsList)
+}
+
+// generate gets the necessary info the create the init secret data
+func (g *BootstrapperInitGenerator) generate(ctx context.Context, dk *dynakube.DynaKube) (map[string][]byte, error) {
+	hostMonitoringNodes, err := g.getHostMonitoringNodes(dk)
+	if err != nil {
+		return nil, err
+	}
+
+	secretConfig, err := g.createSecretConfigForDynaKube(ctx, dk, hostMonitoringNodes)
+	if err != nil {
+		return nil, err
+	}
+
+	agCerts, err := dk.ActiveGateTLSCert(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	trustedCAs, err := dk.TrustedCAs(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	pmcSecret, err := g.preparePMCSecretForBootstrapper(ctx, *dk)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return map[string][]byte{
+		pmc.InputFileName:        pmcSecret,
+		ca.TrustedCertsInputFile: trustedCAs,
+		ca.AgCertsInputFile:      agCerts,
+		curl.InputFileName:       []byte(strconv.Itoa(secretConfig.InitialConnectRetry)),
+		endpoint.InputFileName:   []byte(secretConfig.ApiToken),
```

Comment:
nononononono

Reply:
This was a palceholder I forgot to change 😓 thanks!

https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/8d9deb9183a43f3c0402f76ec0be8738de41bb50